consensus/types/src/light_client_update.rs

```diff
@@ -120,22 +124,40 @@ impl<T: EthSpec> LightClientUpdate<T> {
         // Build finalized header from finalized block
         let finalized_header = BeaconBlockHeader {
             slot: finalized_block.slot(),
-            proposer_index: finalized_block.proposer_index(),
+            proposer_index: finalized_block.message().proposer_index(),
             parent_root: finalized_block.parent_root(),
             state_root: finalized_block.state_root(),
-            body_root: finalized_block.body_root(),
+            body_root: finalized_block.message().body_root(),
         };
         if finalized_header.tree_hash_root() != beacon_state.finalized_checkpoint().root {
             return Err(Error::InvalidFinalizedBlock);
         }
         let next_sync_committee_branch =
             attested_state.compute_merkle_proof(NEXT_SYNC_COMMITTEE_INDEX)?;
         let finality_branch = attested_state.compute_merkle_proof(FINALIZED_ROOT_INDEX)?;
+
+        let (attested_header, finalized_header) =
+            match chain_spec.fork_name_at_epoch(beacon_state.slot().epoch(T::slots_per_epoch())) {
```

Comment:
I think you can just match on `beacon_state` here?

Reply:
i think this one can be deleted too!

Reply:
was under the assumption that the linter would pick up dead code, weird that it didnt pick these ones up

Reply:
yeah i think pub functions in lib crates are allowed and not considered dead code.

Reply:
changed to match on beacon_state